smack-extensions/src/main/java/org/jivesoftware/smackx/avatar/MetadataInfo.java

```diff
@@ -0,0 +1,120 @@
+/**
+ *
+ * Copyright 2017 Fernando Ramirez, 2019 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.avatar;
+
+import java.net.URL;
+
+/**
+ * User Avatar metadata info model class.
+ *
+ * @author Fernando Ramirez
+ * @author Paul Schaub
+ * @see <a href="http://xmpp.org/extensions/xep-0084.html">XEP-0084: User
+ *      Avatar</a>
+ */
+public class MetadataInfo {
+
+    public static final int MAX_HEIGHT = 65536;
+    public static final int MAX_WIDTH = 65536;
+
+    private final String id;
+    private final URL url;
+    private final long bytes;
```

Comment:
Should be of type `UInt16`.

Reply:
Hm, although the XEP states so, I think it should rather be `UInt32` for 'bytes'.
See my [posting to standards@](https://mail.jabber.org/pipermail/standards/2019-August/036376.html) for the rationale.